extensions/amp-bind/0.1/bind-impl.js

```diff
@@ -629,6 +646,43 @@ export class Bind {
   }
 
   /**
+   *
+   */
+  beginObservingElementForMutations_(element) {
+    // TODO(kmh287): What if parent is the body tag?
+    // TODO(kmh287): Generify logic for node observation strategy
+    // when bind supprots more dynamic nodes.
+    const elementToObserve = element.parentElement;
+    this.mutationObserver_.observe(elementToObserve, {childList: true});
+  }
+
+  /**
+   * Respond to observed mutations. Meant to be used with MutationObserver
+   *
+   * @param mutations {Array<MutationRecord>}
+   */
+  onMutationsObserved_(mutations) {
+    mutations.forEach(mutation => {
+      const mutationPromises = [];
+      const removedNodes = mutation.removedNodes;
+      for (let i = 0; i < removedNodes.length; i++) {
```

Comment:
It seems like we could just filter these to the `Element`s (`nodeType == 1`), and keep the type signatures the same (and also perform a lot less operations).